README.md

```diff
@@ -51,6 +51,132 @@ After this, you will have a directory containing files used for creating a
+Marks the `<python_package_name>` directory as a python module, allowing `<python_package_name>` to be imported as a module
+
+    - **`_version.py`**\
+Contains python module version definitions
+
+    - **`example.py`**\
+Widget class definition in python. Defines widget data model that is synced with front-end automatically using traitlets and Backbone.js
+
+- **`js directory`**\
+Source files for widget front-end. Bundled as an npm package\
+
+    - **`lib directory`**\
+Directory containing source files
+
+        - **`embed.js`**\
+Webpack bundler entry point for generating front-end code bundle. The output of this bundle is used for unpkg distribution.
+
+        - **`example.js`**\
+Widget front-end source code. Contains widget model and view class definitions. Widget model is derived from [DOMWidgetModel](https://github.com/jupyter-widgets/ipywidgets/blob/master/packages/base/src/widget.ts) which provides functionality to communicate with the model instance on kernel. Widget view is derived from [DOMWidgetView](https://github.com/jupyter-widgets/ipywidgets/blob/master/packages/base/src/widget.ts) which has a `model` attribute referencing to widget model it is presenting. If a notebook outputs same widget more than once, there is still one model in back-end and one in front-end, only separate views are created.
+
+        - **`extension.js`**\
+Contains the javascript that is run when the notebook is loaded. It contains some requirejs configuration and the `load_ipython_extension` which is required for any notebook extension.
+
+        - **`index.js`**\
+Webpack bundler entry point for generating front-end code bundle.
+
+        - **`labplugin.js`**\
+JupyterLab extension definition for JupyterLab integration. When JpuyterLab loads, it activates the plugin defined in this file. Widget registry service provided by `@jupyter-widgets/jupyterlab-manager` is used during activation to register the widget with JupyterLab.
+
+    - **`package.json`**\
+npm package definiton file for widget front-end
+
+    - **`README.md`**\
+Auto-generated documentation file for front-end npm package, with installation instructions.
+
+    - **`webpack.config.js`**\
+Webpack bundler configuration that defines bundler entry points and ouputs along with other configuration
+
+- **`.gitignore`**\
+Ignore rules for widget project git repository.
+
+- **`<npm_package_name>.json`**\
+nbextension configuration file used by Jupyter Notebook
+
+- **`MANIFEST.in`**\
+setuptools configuration file containing commands to execute (include / exclude files) during installation (pip install).
+
+- **`README.md`**\
+Auto-generated documentation file for widget project.
+
+- **`RELEASE.md`**\
+Auto-generated documentation file for publishing python package to PyPI and npm package to npm registry.
+
+- **`setup.cfg`**\
+Configuration file to mark the python package as a Universal Wheel.
+
+- **`setup.py`**\
+Build script for the widget package.
+
+## Local Dev Installation
```

Comment:
"Local dev installation when developing with the classic notebook"

(the instructions are different for JLab)